packages/flutter/lib/src/widgets/scroll_position.dart

```diff
@@ -15,202 +15,244 @@ import 'package:flutter/scheduler.dart';
+  double applyPhysicsToUserOffset(ScrollPosition position, double offset) {
+    if (parent == null)
+      return offset;
+    return parent.applyPhysicsToUserOffset(position, offset);
+  }
 
   /// Determines the overscroll by applying the boundary conditions.
   ///
-  /// Called by [AbsoluteScrollPosition.setPixels] just before the [pixels] value is updated, to
-  /// determine how much of the offset is to be clamped off and sent to
-  /// [AbsoluteScrollPosition.reportOverscroll].
+  /// Called by [ScrollPosition.setPixels] just before the [pixels] value is
+  /// updated, to determine how much of the offset is to be clamped off and sent
+  /// to [ScrollPosition.reportOverscroll].
   ///
   /// The `value` argument is guaranteed to not equal [pixels] when this is
   /// called.
-  double applyBoundaryConditions(AbsoluteScrollPosition position, double value) => 0.0;
+  double applyBoundaryConditions(ScrollPosition position, double value) {
+    if (parent == null)
+      return 0.0;
+    return parent.applyBoundaryConditions(position, value);
+  }
 
   /// Returns a simulation for ballisitic scrolling starting from the given
   /// position with the given velocity.
   ///
   /// If the result is non-null, the [ScrollPosition] will begin an
-  /// [AbsoluteBallisticScrollActivity] with the returned value. Otherwise, the
+  /// [BallisticScrollActivity] with the returned value. Otherwise, the
   /// [ScrollPosition] will begin an idle activity instead.
-  Simulation createBallisticSimulation(AbsoluteScrollPosition position, double velocity) => null;
+  Simulation createBallisticSimulation(ScrollPosition position, double velocity) {
+    if (parent == null)
+      return null;
+    return parent.createBallisticSimulation(position, velocity);
+  }
 
-  static final SpringDescription _kDefaultScrollSpring = new SpringDescription.withDampingRatio(
+  static final SpringDescription _kDefaultSpring = new SpringDescription.withDampingRatio(
     mass: 0.5,
     springConstant: 100.0,
     ratio: 1.1,
   );
 
-  SpringDescription get scrollSpring => _kDefaultScrollSpring;
-}
-
-abstract class ScrollPhysicsProxy extends ScrollPhysics {
-  const ScrollPhysicsProxy(this.parent);
-
-  final ScrollPhysics parent;
-
-  @override
-  ScrollPhysicsProxy applyTo(ScrollPhysics parent) {
-    throw new FlutterError(
-      '$runtimeType must override applyTo.\n'
-      'The default implementation of applyTo is not appropriate for subclasses '
-      'of ScrollPhysicsProxy because they should return an instance of themselves '
-      'with their parent property replaced with the given ScrollPhysics instance.'
-    );
-  }
-
-  @override
-  double applyPhysicsToUserOffset(AbsoluteScrollPosition position, double offset) {
+  SpringDescription get spring {
     if (parent == null)
-      return super.applyPhysicsToUserOffset(position, offset);
-    return parent.applyPhysicsToUserOffset(position, offset);
+      return _kDefaultSpring;
+    return parent.spring;
```

Comment:
i'm totally fine with this the way you wrote it, but i'm sure you would tell me to use `parent?.spring ?? _kDefaultSpring` :-)